lib/backends/sdc/containers.js

```diff
@@ -559,60 +525,312 @@ function addRulesToFWAPI(opts, rules, payload, callback) {
+        } else {
+            return namedNetwork(opts, container, payload, callback);
+        }
     }
+}
+
+/*
+ * When fabrics are not configured, we will use the named
+ * network, defaulting to 'external'.
+ */
+function externalNetworkByName(opts, payload, callback) {
+    assert.object(opts.config, 'opts.config');
+    assert.optionalString(opts.config.externalNetwork,
+        'opts.config.externalNetwork');
+    assert.object(opts.account, 'opts.account');
+    assert.string(opts.account.uuid, 'opts.account.uuid');
+    assert.object(payload, 'payload');
 
     var log = opts.log;
-    log.debug('Fabrics not configured: using external network');
+    var listParams = {
+        name: opts.config.externalNetwork || 'external',
+        fabric: false,
+        provisionable_by: opts.account.uuid
+    };
 
-    // No fabrics configured - fall back to provisioning on the external
-    // network
+    log.debug({ listParams: listParams },
+        format('Networks: fabrics not configured, using network %s',
+        listParams.name));
 
-    var netName = opts.config.externalNetwork || 'external';
-    listNetworks(opts, {name: netName}, function (err, networks) {
-        var external_net;
-        log.debug({err: err, res: networks}, 'list external networks');
+    listNetworks(opts, listParams, function (err, networks) {
+        log.debug({ err: err, res: networks },
+            format('Networks: listNetworks result for %s', listParams.name));
 
         if (err) {
-            callback(errors.napiErrorWrap(err, 'problem listing networks'));
+            callback(errors.napiErrorWrap(err,
+                format('Networks: problem listing network %s',
+                    listParams.name)));
             return;
         }
 
-        networks.forEach(function (n) {
-            if (!external_net
-                && n.name === netName) {
+        if (networks.length < 1) {
+            log.error({ networks: networks, params: listParams },
+                format('Networks: network %s provisionable by %s not found',
+                    listParams.name, listParams.provisionable_by));
+            callback(new errors.NetworkNotFoundError(listParams.name));
+            return;
+        }
 
-                external_net = n.uuid;
-            }
-        });
+        payload.networks = [ { uuid: networks[0], primary: true} ];
+        return callback();
+    });
+}
 
-        if (!external_net) {
-            callback(new errors.DockerError(
-                'unable to find "'+netName+'" network uuid'));
+/*
+ * When fabrics are configured, but no specific network is supplied,
+ * or if 'bridge' is supplied, we will use the user's default fabric
+ * network, stored in UFDS.
+ *
+ * Additionally, if the user is publishing ports or we have the
+ * `fabricRequireExternal` option (currently used only by docker build,
+ * see DOCKER-705), we will also attach to configured network pool
```

Comment:
I think that `(used by docker build, for example)` would be better, so that if others use it in the future, the parenthetical won't become incorrect.